### PR TITLE
CI overhaul

### PR DESCRIPTION
**Pull request by FynnBe**
todo:
 - [ ] run on push to `main`, not `pyci`
 - [x] figure out how to install conda envs in parallel... (make sure downloaded packages are there somehow, see https://stackoverflow.com/questions/58210335/is-conda-install-a-thread-safe-operation) 
   - -->it just works in parallel?
 - [ ] suport for github resources
 - [x] fix open-prs

### CODE REVIEW

**Comment by oeway** [comment]
@FynnBe Is this ready? I think we can just merge and test it!

File: collection/10.5281/zenodo.5744489/concept.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+concept_doi: 10.5281/zenodo.5744489
```
Comment (constantinpape) [inline collection/10.5281/zenodo.5744489/concept.yaml:1]: What exactly is the difference between the concept_doi and id?

**Review by constantinpape — COMMENTED** [review]
I think we can go ahead as well.
Just conceptually: what's the difference between concept_doi and id?

File: collection/10.5281/zenodo.5744489/concept.yaml (hunk 1 above)
Comment (oeway) [reply collection/10.5281/zenodo.5744489/concept.yaml:1]: They are the same in this case, but id is common for rdfs from zenodo or github, and concept id is only available for zenodo.

**Comment by oeway** [comment]
One minor comment here, can we rename `concept.yaml` to something like `rdf.yaml` to make it also suitable for items from github? I would still call it `rdf.yaml` because that will allow us to use the RDF fields to provide additional annotation to the items.

**Comment by FynnBe** [comment]
> One minor comment here, can we rename `concept.yaml` to something like `rdf.yaml` to make it also suitable for items from github? I would still call it `rdf.yaml` because that will allow us to use the RDF fields to provide additional annotation to the items.

I really don't want to rename it rdf.yaml. You can completely ignore it. once this completely runs, the PRs are adding a `card.yaml` which holds the information for the website card. The `concept.yaml` is just there to keep the state of found resources.
 
IMO `card.yaml` shouldn't be renamed `rdf.yaml` either, because it's not and specific for the website only, the card itself also doesn't have a format_version or something like that, we could just regenerate all of them if we need breaking changes.
 
 support for github resources is not complete yet, but I kept it in mind and will add it soon (that's why we have `id` and `concept_doi` and `version_id` and `doi`, which are identical for zenodo resources. `id` can then be used for github resources with `version_id` as a git commit of that resource.

**Comment by FynnBe** [comment]
> @FynnBe Is this ready? I think we can just merge and test it!

no quite, but the ci is configured such that it runs as if this branch was main...

File: collection/10.5281/zenodo.5744489/concept.yaml (hunk 1 above)
Comment (FynnBe) [reply collection/10.5281/zenodo.5744489/concept.yaml:1]: yes `version_id` and `doi` have an analog relationship

**Comment by oeway** [comment]
> I really don't want to rename it rdf.yaml. You can completely ignore it. once this completely runs, the PRs are adding a card.yaml which holds the information for the website card. The concept.yaml is just there to keep the state of found resources.

> IMO card.yaml shouldn't be renamed rdf.yaml either, because it's not and specific for the website only, the card itself also doesn't have a format_version or something like that, we could just regenerate all of them if we need breaking changes.

Could you clarify which one will sit in the main branch and act as the source of truth for the entire collection? I guess it's the files what you called `concept.yaml`, no?

If that's the case, I don't mean we only name it rdf.yaml but I mean we should make the format compatible as RDF spec. Because that's the thing we will edit, and it would be nice if it can adhere to some defined spec. If we already have RDF spec, why bother make a new one?

For the card.yaml, is that the file with all the information for each item displayed on the website? I don't see why we want to create yet another format for this. Wouldn't it just be better if we add format_version etc. to make it an RDF file?

My point is that if there is no reason to create a new format, we should stick with the RDF spec.

File: collection/10.5281/zenodo.5744489/concept.yaml (hunk 1 above)
Comment (constantinpape) [reply collection/10.5281/zenodo.5744489/concept.yaml:1]: Ok, thanks for the clarification (should be documented somewhere, but otherwise fine with me)

**Comment by constantinpape** [comment]
> My point is that if there is no reason to create a new format, we should stick with the RDF spec.

If I understand @FynnBe correctly, the point is that `rdf.yaml` would imply that it fully adheres to the rdf spec (in the sense of passing `bioimageio validate`) and we don't want to restrict ourselves to this here because we are just dealing with internals of the website.

**Comment by FynnBe** [comment]
> Could you clarify which one will sit in the main branch and act as the source of truth for the entire collection? I guess it's the files what you called concept.yaml, no?

no there will be `card.yaml` after a PR has been merged (the CI didn't get to create a PR yet.. I'm fixing those bugs..)

**Comment by oeway** [comment]
> > My point is that if there is no reason to create a new format, we should stick with the RDF spec.
> 
> If I understand @FynnBe correctly, the point is that `rdf.yaml` would imply that it fully adheres to the rdf spec (in the sense of passing `bioimageio validate`) and we don't want to restrict ourselves to this here because we are just dealing with internals of the website.

I understand, but I am not sure that's entirely internals of the website. The `concept.yaml` in the main branch is interface for us to approve items and add additional information (e.g. tags & badges from our admin team ), and the `card.yaml` will be rendered on the website and the website assumes the format is RDF compatible. There are also tools such as deepimagej model downloader implementation will use these files to render a model list in deep imagej.

To make the coordination between repos and our internal members easier, I think we should simply say all these are RDF, so we can avoid extra effort for coordination.

Plus, it's really easy to pass the general RDF validator, it has only a few mandatory fields and allows additional properties.

**Comment by FynnBe** [comment]
the rdf has no "previous_versions" and no "id" and no "doi" fields... there are already differences.

I think I haven't sufficiently explained this new CI... maybe we can talk about it, that would make things easier...

**Comment by oeway** [comment]
> the rdf has no "previous_versions" and no "id" and no "doi" fields... there are already differences.
> 
> I think I haven't sufficiently explained this new CI... maybe we can talk about it, that would make things easier...

Well, because we allow additional fields in the RDF spec, those new fields won't make the choice of RDF format invalid. What matters the RDF spec give us a way to add fields that are already defined in the spec, the website will make sure those are displayed correctly.

Yes, let's discuss this when we meet.

**Comment by oeway** [comment]
Hey, just want to say, the choice of spec we are discussing here is rather minor, and it's easy enough to change afterwards. For now I would simply focus on a working version and try to get it work end-to-end, feel free to merge when you get it work.

There are a few other things for the partners and bioengine apps I need to fix later, but I will wait for this PR.

**Comment by FynnBe** [comment]
almost there... still fixing https://github.com/bioimage-io/collection-bioimage-io/actions/runs/1580246071

**Comment by constantinpape** [comment]
Very nice to see this merged :)

**Comment by FynnBe** [comment]
first batch of generated PRs: #34, #35, #37

**Pull request merged**